### PR TITLE
Make cf push work from outside

### PR DESCRIPTION
**Pull request by phil9909**
This PR makes `cf push` work from outside the cluster. All traffic coming from outside is routed over the ingress which requires `features.ingress.enabled=true`.

## Description

* This PR primarily bumps the version of eirini from 0.0.14 to 0.0.17.
* In turn some additional changes were required as eirini now is https only.
* The blobstore was only accessible for a small set of IP-Addresses.
* Some patches were fixed for ease of debugging. See #2844. These changes are not required. We could move them out of this PR.
* The certificate for `registry.(( system domain))` needs to be signed by the `internalCA` not `kubernetes`. Otherwise the router will not be able to forward requests.

## Motivation and Context
#2845 
#2867

## How Has This Been Tested?
`cf push`

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

Maybe requiring ingress implies "My change requires a change to the documentation"?

### CODE REVIEW

File: deploy/helm/scf/assets/operations/instance_groups/bits.yaml [hunk 1]
```diff
@@ -31,7 +31,7 @@
     namespace: {{ .Release.Namespace }}
 - type: replace
   path: /variables/name=bits_service_ssl/options/signer_type?
-  value: cluster
+  value: local
```
Comment (viovanov) [inline deploy/helm/scf/assets/operations/instance_groups/bits.yaml]: Can you please explain this?
From our tests, using a Kube CSR was the only way to get the nodes to trust our registry.

Comment (phil9909) [reply deploy/helm/scf/assets/operations/instance_groups/bits.yaml]: * In order to make push work from the outside the bit-service must be reachable, which is achieved via the `gorouter`. But the `gorouter` does not trust the `cluster` CA.
* To keep it possible for the `kubelet` to pull the image, we changed it to pull the image using the external url (i.e. it also goes thru the `gorouter`).

Comment (viovanov) [reply deploy/helm/scf/assets/operations/instance_groups/bits.yaml]: I don't think your first statement is correct. The bits service should _only_ be used as a registry for Eirini docker images.
Pulling the image through the gorouter means the Kube nodes need to trust its CA - how does that happen?

**Comment by viovanov** [comment]
@phil9909 going through the PR again - I don't understand what the title has to do with your changes.
Also, please explain why you're making registry traffic pass through the router. We're specifically trying to avoid this. We want the registry to be exposed using a cert signed by Kube so it's trusted by nodes.

**Comment by phil9909** [comment]
> @phil9909 going through the PR again - I don't understand what the title has to do with your changes.
> Also, please explain why you're making registry traffic pass through the router. We're specifically trying to avoid this. We want the registry to be exposed using a cert signed by Kube so it's trusted by nodes.

I see your point (now).
How is the kubelet able to access the bits-service / image-registry inside the cluster without an external IP / DNS? We did not understand this, but if you figured out how to do this we are happy to rework this PR.

**Comment by phil9909** [comment]
I went further down the rabbit hole and finally figured out the root cause of the confusion here.
Clusters provided by the Google Cloud Platform (i.e. GKE clusters) are different from the clusters we use ([Gardener](https://github.com/gardener/gardener/) provided clusters) in the following way: the docker-daemon which runs on the nodes of GKE clusters trusts the cluster CA, which is not the case on Gardener Clusters.
This is why we weren't able to get things running and thought this change would be required. I have opened an issue at the Gardener project https://github.com/gardener/gardener/issues/1515 and they will look into this soon. I have no idea if the current implementation also breaks on other clusters providers.

> Pulling the image through the gorouter means the Kube nodes need to trust its CA - how does that happen?

As the gorouter in our setup has a valid (let's encrypt) certificate this is a no-brainer. I understand the for some setups (mostly dev and CI setups) this won't be true.

**Comment by f0rmiga** [comment]
@phil9909 I pushed a commit to fix the Eirini addresses that I removed when switching to use the BOSH DNS aliases and a commit to add `-o errexit -o nounset` to the configure-eirini errand.

**Comment by f0rmiga** [comment]
@phil9909 Could you reopen this PR on github.com/SUSE/kubecf, please?

**Comment by kramerul** [comment]
@f0rmiga  I opened two new PRs on [SUSE/kubecf](https://github.com/SUSE/kubecf) derived from this one
* [Bump eirini version in scf to 0.0.21](https://github.com/SUSE/kubecf/pull/59)
* [Support additional cidrs for k8s clusters](https://github.com/SUSE/kubecf/pull/58)

**Pull request closed without merging**